### ISSUE TITLE
ion-footer on back should disappear on willLeave not didLeave

### ISSUE THREAD

**Issue by peterpeterparker** [driftyco/ionic#10680]
**Ionic version:**  (check one with "x")
[ X] **2.x**

**I'm submitting a ...**  (check one with "x")
[X ] bug report

**Current behavior:**
The ion-footer of sub-pages (accessed thru navController.push) seems to disappear, when going back to the previous page, only once the view did effectively leave. 

If you apply a background-color to the ion-footer, that lead to a weird effect, where, after the click on the back button, the previous page is displayed and one second later the footer blink while disappearing.

**Expected behavior:**
The ion-footer should disappear on vill leave or follow the same effect as the page ("slowly disappearing to the right")

**Steps to reproduce:**

See video: https://dl.dropboxusercontent.com/u/77620770/back_ion-footer.mov

**Related code:**

    <ion-footer style="background-color: red">
         <button ion-button round (click)="stuff()">Test</button>
    </ion-footer>

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

Your system information:

Cordova CLI: 6.5.0 
Ionic Framework Version: 2.1.0
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.1.4
ios-deploy version: 1.9.0 
ios-sim version: 5.0.13 
OS: macOS Sierra
Node Version: v7.2.1
Xcode version: Xcode 8.2.1 Build version 8C1002

**Comment by peterpeterparker** [comment]
Ok I found a solution, if I add a ion-toolbar and apply the background-color to that one (not to the all footer), then the transition on back looks ok and solve that issue

    <ion-footer>
        <ion-toolbar  style="background-color: red">
             <button ion-button round (click)="stuff()">Test</button>
        </ion-toolbar>
    </ion-footer>

**Comment by dani-inside** [comment]
+1

**Comment by peterpeterparker** [comment]
@dani-inside maybe a workaround or maybe it's designed like this don't know. 

If the solution I found didn't work for you, double check that you don't have a ngIf or other angular execution in that footer part which could trigger the glitch too. If yes, move it to the above or under element and it should work (it was my case).

if still doesn't work, maybe you could a open a new issue, I'm fine with this solution for the moment at least.

**Comment by dani-inside** [comment]
Thanks for the feedback @peterpeterparker, appreciate it. It's actually not solving the issue for me, and there are no angular directives on the elements. I'll open a new ticket then.

**Comment by haifzhan** [comment]
@dani-inside Have the same issue here...any solutions now?
